### PR TITLE
add global var file & resolver

### PR DESCRIPTION
**Pull request by ecobrien29**
These changes add:
1. a GlobalVar resolver that gets the value on the variable from the globalParameterFile
2. reading in global parameters from the globalParameterFile and merging them with the region & local parameters

Also trying to update our really outdated ruby dependencies. can separate if needed

### CODE REVIEW

File: lib/stacker/resolvers/global_var_resolver.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+require 'stacker/resolvers/resolver'
+
+module Stacker
+  module Resolvers
+
+    class GlobalVarResolver < Resolver
+
+      def resolve
+        global_defaults = region.defaults.fetch 'globalParameterFile', {}
+        global_vars = YAML.load_file(global_defaults)
```
Comment (xiaominforward) [inline lib/stacker/resolvers/global_var_resolver.rb:10]: do we need to check if `global_defaults` is empty before calling `load_file`?

File: lib/stacker/stack/parameters.rb [hunk 2]
```diff
@@ -18,9 +18,19 @@ def region_defaults
...
       end
 
+      def global_defaults
+        stack.region.defaults.fetch 'globalParameterFile', {}
+      end
+
       # template defaults merged with region and stack-specific overrides
       def local
         region_defaults = stack.region.defaults.fetch 'parameters', {}
+        global_defaults = stack.region.defaults.fetch 'globalParameterFile', {}
```
Comment (xiaominforward) [inline lib/stacker/stack/parameters.rb]: this line is probably not needed. the line above probably not needed either

File: lib/stacker/stack/parameters.rb [hunk 3]
```diff
@@ -31,6 +41,8 @@ def local
         ]
 
         available = template_defaults.merge(
+          global_vars
+        ).merge(
```
Comment (xiaominforward) [inline lib/stacker/stack/parameters.rb:43]: does the order of merging matters here? if some parameter has one value in the global variable file, but another value in the region file, what value would it take? from the infra PR, looks like we are keeping `parameters` section in the region file? I'm worried about having more than once place for global parameters.

Comment (ecobrien29) [reply lib/stacker/stack/parameters.rb:43]: yes, the region variables will override the global variables. it made more sense for general usecase and hierarchy, even though it's not great for enforcing on our side. I'm going to add comments to our yaml file saying which is regional vs global

File: lib/stacker/resolvers/global_var_resolver.rb (hunk 1 above)
Comment (ecobrien29) [reply lib/stacker/resolvers/global_var_resolver.rb:10]: if they're using this then it's assumed the global var file exists, kind of like how we handle files

**Pull request closed without merging**